test/unit/core/type_traits/basic_test.cpp

```diff
@@ -34,3 +38,73 @@ TEST(type_trait, remove_cvref_t)
     EXPECT_FALSE((std::is_same_v<int*, seqan3::remove_cvref_t<int[3]>>));   // type stays same
     // the last example would be true for std::decay_t
 }
+
+//------------------------------------------------------------------------------
+// Tests for is_constexpr
+//------------------------------------------------------------------------------
+
+constexpr
+int    constexpr_nonvoid_free_fun(int i) { return i; }
+int nonconstexpr_nonvoid_free_fun(int i) { return i; }
+
+constexpr
+int constexpr_nonvoid_free_fun_const_ref(int const & i) { return i; }
+int nonconstexpr_nonvoid_free_fun_const_ref(int const & i) { return i; }
+
+constexpr
+void    constexpr_void_free_fun(int) { return; }
+void nonconstexpr_void_free_fun(int) { return; }
+
+struct constexpr_nonvoid_member_t
+{
+    int constexpr get_i(int i)
+    {
+        return i;
+    }
+};
+
+struct constexpr_void_member_t
+{
+    void constexpr get_i(int)
+    {}
+};
+
+struct nonconstexpr_nonvoid_member_t
+{
+    int get_i(int i)
```

Comment:
can't this function also live in the `constexpr_nonvoid_member_t` struct?

Reply:
Dunno. Seems to work. Won't try.